phoebe/atmospheres/passbands.py

```diff
@@ -724,47 +725,64 @@ def compute_blackbody_intensities(self, teffs=None, include_extinction=False, rv
+            pgrid = np.empty(shape=(len(teffs), len(ebvs), len(rvs), 1))
+            egrid = np.empty(shape=(len(teffs), len(ebvs), len(rvs), 1))
 
-            egrid = (extincted_intensities_energy/non_extincted_intensities_energy).reshape(len(teffs), len(rvs), len(ebvs), 1)  # (97, 16, 30, 1)
-            pgrid = (extincted_intensities_photon/non_extincted_intensities_photon).reshape(len(teffs), len(rvs), len(ebvs), 1)  # (97, 16, 30, 1)
+            for ti, teff in enumerate(teffs):
+                bb_sed = self._planck(self.wl, teff)
+                for ei, ebv in enumerate(ebvs):
+                    for ri, rv in enumerate(rvs):
+                        Alam = 10**(-0.4 * ebv * (rv * ax + bx))
+                        egrid[ti, ei, ri, 0] = np.trapz(self.ptf(self.wl) * bb_sed * Alam, axis=0) / np.trapz(self.ptf(self.wl) * bb_sed, axis=0)
+                        pgrid[ti, ei, ri, 0] = np.trapz(self.wl * self.ptf(self.wl) * bb_sed * Alam, axis=0) / np.trapz(self.wl * self.ptf(self.wl) * bb_sed, axis=0)
 
             self.ndp['blackbody'] = ndpolator.Ndpolator(basic_axes=(axes[0],))
             self.ndp['blackbody'].register('ext@photon', associated_axes=(axes[1], axes[2]), grid=pgrid)
             self.ndp['blackbody'].register('ext@energy', associated_axes=(axes[1], axes[2]), grid=egrid)
 
+            # ebv_column = np.tile(ebvs, len(rvs))
+            # rvebv_column = ebv_column*np.repeat(rvs, len(ebvs))
+            # ext_pars = np.vstack((rvebv_column, ebv_column))
+            # ext_func = libphoebe.gordon_extinction(wls)  # (47, 2)
+            # ext = ext_func @ ext_pars  # (47, 480)
+            # flux_fracs = 10**(-0.4*ext)  # (47, 480)
+            # integrand_energy = (pbpfs_energy[:, None, :]*flux_fracs[:, :, None]).T  # ~25ms  (97, 480, 47)
+            # integrand_photon = (pbpfs_photon[:, None, :]*flux_fracs[:, :, None]).T  # ~25ms  (97, 480, 47)
+
+            # extincted_intensities_energy = np.trapz(integrand_energy, self.wl, axis=2)  # (97, 480)
+            # extincted_intensities_photon = np.trapz(integrand_photon, self.wl, axis=2)  # (97, 480)
+
+            # non_extincted_intensities_energy = np.trapz(pbpfs_energy, self.wl, axis=0)[:,None]  # (97, 1)
+            # non_extincted_intensities_photon = np.trapz(pbpfs_photon, self.wl, axis=0)[:,None]  # (97, 1)
+
+            # egrid = (extincted_intensities_energy/non_extincted_intensities_energy).reshape(len(teffs), len(ebvs), len(rvs), 1)  # (97, 16, 30, 1)
+            # pgrid = (extincted_intensities_photon/non_extincted_intensities_photon).reshape(len(teffs), len(ebvs), len(rvs), 1)  # (97, 16, 30, 1)
+
+            # self.ndp['blackbody'] = ndpolator.Ndpolator(basic_axes=(axes[0],))
+            # self.ndp['blackbody'].register('ext@photon', associated_axes=(axes[1], axes[2]), grid=pgrid)
+            # self.ndp['blackbody'].register('ext@energy', associated_axes=(axes[1], axes[2]), grid=egrid)
+
```

Comment:
Let's leave these in for now because Dave's workaround increases the time cost by an order of magnitude. It'd be better to fix this code than to rely on the much slower implementation. I'll remove the comments before the final PR.